flight_planning/v1/flight_planning.yaml

```diff
@@ -0,0 +1,371 @@
+          description: >-
+            The USS automated testing interface is available and its status was retrieved successfully.
+        '401':
+          description: Bearer access token was not provided in Authorization header, token could not be decoded, or token was invalid.
+        '403':
+          description: The access token was decoded successfully but did not include a scope appropriate to this endpoint.
+        '404':
+          description: The USS automated testing interface is not available.
+      summary: Status of the USS automated testing interface
+      description: Get the status of the USS automated testing interface.
+
+  /clear_area_requests:
+    post:
+      security:
+        - Authority:
+            - interuss.flight_planning.direct_automated_test
+
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ClearAreaRequest'
+        required: true
+      responses:
+        '200':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ClearAreaResponse'
+          description: Requested area was cleared successfully
+        '401':
+          description: Bearer access token was not provided in Authorization header, token could not be decoded, or token was invalid.
+        '403':
+          description: The access token was decoded successfully but did not include a scope appropriate to this endpoint.
+
+      summary: Clear area
+      operationId: ClearArea
+      description: >-
+        This endpoint requests that the USS cancel and remove all flight plans in the specified area.
+
+  /flight_plans/{flight_plan_id}:
+    parameters:
+      - name: flight_plan_id
+        in: path
+        required: true
+        description: A UUID string identifying the user's flight plan intent.
+        schema:
+          $ref: '#/components/schemas/UUIDv4Format'
+
+    put:
+      security:
+        - Authority:
+            - interuss.flight_planning.plan
+
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UpsertFlightPlanRequest'
+        required: true
+      responses:
+        '200':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UpsertFlightPlanResponse'
+          description: Requested data was processed successfully
+        '401':
+          description: Bearer access token was not provided in Authorization header, token could not be decoded, or token was invalid.
+        '403':
+          description: The access token was decoded successfully but did not include a scope appropriate to this endpoint.
+
+      summary: Upsert flight plan
+      operationId: UpsertFlightPlan
+      description: >-
+        This endpoint simulates an operator intention to submit a new or updated flight plan.
+
+    delete:
```

Comment:
Is this call idempotent? If yes: could be useful to clarify it returns 200 if the flight does not exist; if not it could make sense to add a `404`.

Reply:
This call is not idempotent.  My thought was that this basically represents a user interaction and users don't follow nice flow control patterns.  It probably would make sense to add a `request_id` to make it possible to de-duplicate incoming requests though; I'll do that (also added a 409 response).  Please take a look at the new commit when pushed.

I'm not sure I follow the 404; I would expect an upsertion operation to either create or update a flight.  If the flight didn't already exist, then it's a creation request.  If the flight did already exist, it's a mutation request.  I would expect a 200 in either case (or, arguably a 201 in the creation case).

Reply:
> I'm not sure I follow the 404; I would expect an upsertion operation to either create or update a flight. If the flight didn't already exist, then it's a creation request. If the flight did already exist, it's a mutation request. I would expect a 200 in either case (or, arguably a 201 in the creation case).

That wasn't clear, my bad, my comment referred to the `delete` call. Thus why returning 404 when attempting to delete an already deleted flight plan kinda made sense to me.

Reply:
> It probably would make sense to add a request_id to make it possible to de-duplicate incoming requests though; I'll do that (also added a 409 response). Please take a look at the new commit when pushed.

LGTM :+1: 
For the sake of consistency, should the same be done for the delete call?

Reply:
Ah, 404 for delete makes sense -- added.  Delete is still idempotent without a request_id so I think it's probably not necessary for delete.